app/models/katello/subscription_status.rb

```diff
@@ -0,0 +1,54 @@
+module Katello
+  class SubscriptionStatus < HostStatus::Status
+    UNKNOWN = 3
+    INVALID = 2
+    PARTIAL_COMPLIANCE = 1
+    FULL_COMPLIANCE = 0
+
+    def self.status_name
+      N_("Subscription Status")
+    end
+
+    def to_label(_options = {})
+      case status
+      when FULL_COMPLIANCE
+        N_("Fully entitled")
+      when PARTIAL_COMPLIANCE
+        N_("Partially entitled")
+      when INVALID
+        N_("Unentitled")
+      else
+        N_("Unknown subscription status")
+      end
+    end
+
+    def to_global(_options = {})
+      case status
+      when INVALID
+        ::HostStatus::Global::ERROR
+      when PARTIAL_COMPLIANCE
+        ::HostStatus::Global::WARN
+      when FULL_COMPLIANCE
+        ::HostStatus::Global::OK
+      end
+    end
+
+    def to_status(_options = {})
+      return UNKNOWN unless host.subscription_aspect.try(:uuid)
+      case Katello::Candlepin::Consumer.new(host.subscription_aspect.uuid).entitlement_status
```

Comment:
If we are showing a list of N hosts, and their global status, and that includes the subscription status - this would imply we are making N-calls to Candlepin everytime we show a list of hosts?